torch/_refs/nn/functional/__init__.py

```diff
@@ -1166,6 +1166,54 @@ def pdist(a: TensorLikeType, p: float = 2) -> TensorLikeType:
     return t.flatten().index_select(0, i[0] * t.shape[0] + i[1])
 
 
+@register_decomposition(aten.pixel_shuffle)
+@out_wrapper()
+def pixel_shuffle(self: Tensor, upscale_factor: int):
+    batch = self.shape[:-3]
+    C_out = self.shape[-3] // upscale_factor**2
+    HW_out = (self.shape[-2] * upscale_factor, self.shape[-1] * upscale_factor)
+    n = len(batch)
+    return (
+        self.view(
+            *batch,
+            C_out,
+            upscale_factor,
+            upscale_factor,
+            self.shape[-2],
+            self.shape[-1],
+        )
+        .transpose(n + 1, n + 3)
+        .transpose(n + 2, n + 3)
+        .transpose(n + 3, n + 4)
```

Comment:
NIT: This would be more readable as:
```python
B_dims = range(n)
C_dim, r1_dim, r2_dim, H_dim, W_dim = range(n, self.ndim())

...
        .permute(*B_dims, C_dim, H_dim, r1_dim, W_dim, r2_dim)
```

Same trick could be used for `unshuffle`.